C/common/filter_pipeline.cpp

```diff
@@ -405,3 +410,192 @@ void FilterPipeline::completeBranch()
 		m_branchActivations.notify_all();
 	}
 }
+
+/**
+ * Attach the debugger to the pipeline elements
+ */
+bool FilterPipeline::attachDebugger()
+{
+	bool rval =  attachDebugger(m_filters);
+	setDebuggerBuffer(1);
+	return rval;
+}
+
+/**
+ * Attach the debugger to the pipeline elements
+ */
+bool FilterPipeline::attachDebugger(const vector<PipelineElement *>& pipeline)
+{
+	bool ret = true;
+	for (auto& elem : pipeline)
+	{
+		if (!elem->attachDebugger())
```

Comment:
If _PipelineElement::attachDebugger_ fails, the code will continue to try attaching to branch elements. Is that correct?